examples/seismic/model.py

```diff
@@ -50,61 +50,72 @@ class Model(object):
     :param m: The square slowness of the wave
     :param damp: The damping field for absorbing boundarycondition
     """
-    def __init__(self, origin, spacing, vp, nbpml=0, dtype=np.float32,
-                 rho=None, epsilon=None, delta=None, theta=None, phi=None):
+    def __init__(self, origin, spacing, shape, vp, nbpml=0, dtype=np.float32,
```

Comment:
I'm not sure why we need to make this explicit? The information is given by `vp.shape`, so why not pull it from there? Forcing the user to supply this explicitly only makes sense if we were to make `vp` optional, or am I missing a use case here?